demonstrations/tutorial_qgrnn.py

```diff
@@ -0,0 +1,609 @@
+# in a tractable number of optimization steps.
+# The first list represents the :math:`ZZ` interaction parameters and 
+# the second list represents the single-qubit `Z` parameters.
+# 
+# Finally, 
+# we use this information to generate the matrix form of the
+# Ising model Hamiltonian in the computational basis:
+#
+
+
+def create_hamiltonian_matrix(n, graph, params):
+
+    matrix = np.zeros((2 ** n, 2 ** n))
+
+    # Creates the interaction component of the Hamiltonian
+    for count, i in enumerate(graph.edges):
+        m = 1
+        for j in range(0, n):
+            if i[0] == j or i[1] == j:
+                m = np.kron(m, qml.PauliZ.matrix)
+            else:
+                m = np.kron(m, np.identity(2))
+        matrix = np.add(matrix, params[0][count] * m)
+
+    # Creates the bias components of the matrix
+    for i in range(0, n):
+        m1 = m2 = 1
+        for j in range(0, n):
+            if j == i:
+                m1 = np.kron(m1, qml.PauliZ.matrix)
+                m2 = np.kron(m2, qml.PauliX.matrix)
+            else:
+                m1 = np.kron(m1, np.identity(2))
+                m2 = np.kron(m2, np.identity(2))
+        matrix = np.add(matrix, np.add(params[1][i] * m1, m2))
+
+    return matrix
+
+# Prints a visual representation of the Hamiltonian matrix
+ham_matrix = create_hamiltonian_matrix(qubit_number, ising_graph, matrix_params)
+plt.matshow(ham_matrix)
+plt.show()
+
+
+######################################################################
+# Preparing Quantum Data
+# ^^^^^^^^^^^^^^^^^^^^^^
+#
+
+
+######################################################################
+# The collection of quantum data needed to run the QGRNN has two componenets: 
+# copies of a low-energy state, and a collection of time-evolved state, each of which are
+# simply the low-energy state evolved to different times. 
+# For the target Hamiltonian we 
+# defined, we find that a low-energy state is given by the following vector
+# (represented by a Numpy array):
+#
+
+low_energy_state = np.array([-0.02086666+0.00920016j, -0.00379192-0.00859852j,  0.06594626-0.02907913j,
+  0.0119852 +0.02717445j,  0.07633593-0.03366391j,  0.01387486+0.03145572j,
+ -0.24124938+0.10640219j, -0.04385454-0.09941154j,  0.06397641-0.02820407j,
+  0.01162454+0.02636273j, -0.20218878+0.08914518j, -0.03674192-0.08331586j,
+ -0.23404312+0.10320031j, -0.04253486-0.09644206j,  0.73966164-0.32618732j,
+  0.13444079+0.30479209j])
+
+
+######################################################################
+# We can verify that this is a low-energy
+# state by numerically finding the lowest eigenvalue of the Hamiltonian
+# matrix and comparing it to the energy expectation of the low-energy state:
+#
+
+# Finds the energy expectation
+
+def expectation_value(vector, matrix):
+    return np.inner(np.conj(vector), (matrix @ vector))
+
+energy_exp = expectation_value(low_energy_state, ham_matrix)
```

Comment:
```suggestion
energy_exp = np.inner(np.conj(low_energy_state), (ham_matrix @ low_energy_state))
```
and remove the function definition above. Unless it's used later? I haven't read the whole tutorial yet